src/crimson/mgr/client.cc

```diff
@@ -58,31 +58,47 @@ seastar::future<> Client::ms_dispatch(crimson::net::Connection* conn,
   }
 }
 
+seastar::future<> Client::ms_handle_connect(crimson::net::ConnectionRef c)
+{
+  if (conn == c) {
+    // ask for the mgrconfigure message
+    auto m = ceph::make_message<MMgrOpen>();
+    m->daemon_name = local_conf()->name.get_id();
+    return conn->send(std::move(m));
+  } else {
+    return seastar::now();
+  }
+}
+
 seastar::future<> Client::ms_handle_reset(crimson::net::ConnectionRef c)
 {
   if (conn == c) {
-    conn = nullptr;
-    tick_timer.cancel();
+    report_timer.cancel();
```

Comment:
The timer is cancelled here, but seems it won't be rearmed after reconnected unless re-configured.

Plus, I see only `ms_handle_reset()` is calling `reconnect()`, and only `reconnect()` is calling  `msgr.connect(peer, CEPH_ENTITY_TYPE_MGR)`, then who will establish the first connection?

Reply:
> The timer is cancelled here, but seems it won't be rearmed after reconnected unless re-configured.
> 

see https://github.com/ceph/ceph/pull/33797#discussion_r390095871

> Plus, I see only `ms_handle_reset()` is calling `reconnect()`, and only `reconnect()` is calling `msgr.connect(peer, CEPH_ENTITY_TYPE_MGR)`, then who will establish the first connection?

`Client::handle_mgr_map()` will.